docs/extensions/oauth2-flow.md

```diff
@@ -0,0 +1,110 @@
+---
+
+You may need users to authenticate from your Docker Extension using OAuth 2.0 via the web browser, and return them back into the Docker Extension.
+
+> ** Note **
+>
+> This page assumes that you already have an identity provider (IdP) which handles the authentication process and returns an access token, like Google, Azure AD or Okta, among many others.
+
+## Introduction
+
+In OAuth 2.0, the term "grant type" refers to the way an application gets an access token. Although OAuth 2.0 defines several grant types, this page will only describe how to authorize users from your Docker Extension using the Authorization Code grant type.
+
+## Authorization Code Grant Flow
+
+The Authorization Code grant type is used by confidential and public clients to exchange an authorization code for an access token.
+
+After the user returns to the client via the redirect URL, the application will get the authorization code from the URL and use it to request an access token.
+
+![oauth2-flow](./images/oauth2-flow.png){: style=width:80% }
+
+- The Docker Extension makes a request for the user to authorize access to their data.
+- If the user grants access, the Docker Extension then requests an access token from the service provider, passing the access grant from the user and - authentication details to identify the client.
+- The service provider validates these details and returns an access token.
+- The Docker Extension uses the access token to request the user data with the service provider.
+
+## OAuth 2.0 terminology
+
+- **Auth URL** - The endpoint for the API provider authorization server, to retrieve the auth code.
+- **Redirect URI** - The client application callback URL to redirect to after auth. This must be registered with the API provider.
+
+Once the user enters the username and password, he would have been successfully authenticated.
+
+## Open a browser page to authenticate the user
+
+From the extension UI, you can provide a button that, on click, it opens a browser window to authenticate the user.
+
+Use the [ddClient.host.openExternal](https://docs.docker.com/desktop/extensions-sdk/dev/api/dashboard/#open-a-url) API to open a browser to the **auth URL**, for instance:
+
+```typescript
+window.ddClient.openExternal("https://authorization-server.com/authorize?
+  response_type=code
+  &client_id=T70hJ3ls5VTYG8ylX3CZsfIu
+  &redirect_uri=${REDIRECT_URI}
+  &scope=photo+offline_access
+  &state=kH_0FdAtjCfYjOkF);
+```
+
+## Get the authorization code and access token
+
+### From the extension UI
+
+> **Passing the access token**
+>
+> At the moment of this writing, passing the authorization code as a query parameter to the `docker-desktop://dashboard/open` URL is not supported.
+> {: .warning}
+
+You will not be able to get the authorization code from the extension UI by listing `docker-desktop://dashboard/open` as the `redirect_uri` in the OAuth app you're using and concatenating the authorization code as a query parameter.
```

Comment:
Still, we should mention here it's still worth invoking `docker-desktop://dashboard/open` in order to give the focus back to the Docker Desktop app at the end of the login flow (even if the token is exchanged in another way)

Reply:
Sorry, I'm not sure I understand why that info should be placed at this level. 

It was added at the very end of the docs: https://github.com/docker/extensions-sdk/pull/194/files#diff-b419fc886f34a404aa7e1cdbe1c8db08101ad195065697f2eb13a68140de6434R99

LMK if I'm missing something 😄